container/go/cmd/puller/puller.go

```diff
@@ -120,6 +122,56 @@ func pull(imgName, dstPath, format, cachePath string, platform v1.Platform) erro
 		}
 	}
 
+	if err := generateSym(img, dstPath); err != nil {
+		return errors.Wrapf(err, "failed to generate symbolic links to pulled image at %s", dstPath)
+	}
+
+	return nil
+}
+
+// generateSym creates symbolic links to the config.json and .tar.gz layers
+// for use with container_import rule.
+func generateSym(img v1.Image, dstPath string) error {
+	targetDir := path.Join(dstPath, "blobs/sha256")
+	symlinkDir := path.Join(filepath.Dir(dstPath), "image")
+
+	// symlink for config.json.
+	var config v1.Hash
+	var err error
+	if config, err = img.ConfigName(); err != nil {
+		return errors.Wrapf(err, "failed to get the config file's hash information for image")
+	}
+	configDir := path.Join(targetDir, config.Hex)
+	dstLink := path.Join(symlinkDir, "config.json")
+	if _, err := ospkg.Lstat(dstLink); err == nil {
+		ospkg.Remove(dstLink)
+	}
+	if err := ospkg.Symlink(configDir, dstLink); err != nil {
+		return errors.Wrapf(err, "failed to create symbolic link to config.json at %s", configDir)
+	}
+
+	// symlink for the layers.
+	var layers []v1.Layer
+	if layers, err = img.Layers(); err != nil {
+		return errors.Wrapf(err, "failed to get the layers for image")
+	}
+	var layersDir string
+	for i, layer := range layers {
+		var layerDigest v1.Hash
+		if layerDigest, err = layer.Digest(); err != nil {
+			return errors.Wrapf(err, "failed to get layer digest")
+		}
+		layersDir = path.Join(targetDir, layerDigest.Hex)
+		out := strconv.Itoa(i) + ".tar.gz"
+		dstLink = path.Join(symlinkDir, out)
+		if _, err := ospkg.Lstat(dstLink); err == nil {
+			ospkg.Remove(dstLink)
+		}
+		if err = ospkg.Symlink(layersDir, dstLink); err != nil {
+			return errors.Wrapf(err, "failed to create symbolic link to layer")
```

Comment:
return errors.Wrapf(err, "failed to create symbolic link to layer at %s", layerDir)